lib/project_types/script/commands/create.rb

```diff
@@ -33,6 +37,13 @@ def self.help
         allowed_values = Script::Layers::Application::ExtensionPoints.types.map { |type| "{{cyan:#{type}}}" }
         ShopifyCli::Context.message('script.create.help', ShopifyCli::TOOL_NAME, allowed_values.join(', '))
       end
+
+      private
+
+      def name_valid?(name)
+        illegal_characters = ['<', '>', '\\', '/']
+        !name.empty? && !illegal_characters.any? { |e| name.chars.include?(e) }
```

Comment:
I imagine there are other characters which could also be problematic, so perhaps this should be an 'allowed' set, e.g. a-z, digits, hyphens and underscores?